lint/checkstyle.bzl

```diff
@@ -82,6 +83,15 @@ def _checkstyle_aspect_impl(target, ctx):
         return []
 
     files_to_lint = filter_srcs(ctx.rule)
+    config_file = ctx.file._config
+    if ctx.attr._configs:
+        label = str(target.label)
+        keys = sorted(ctx.attr._configs.keys(), key=len, reverse = True)
+        for key in keys:
+            if label.startswith(key):
+                config = ctx.attr._configs[key]
+                config_file = config.files.to_list()[0]
+                break
```

Comment:
do you want longest matching prefix?

Reply:
I'd expect it to use the config from the nearest ancestor, matching most other tools

Reply:
How about a prefactoring where you move your checkstyle configs closer to the folder they govern? Having them all in a single directory seems like it makes it hard to reason about, even without taking Bazel into account. If someone wanted to update the checkstyle config for their code, how would they figure out which one to edit?